app/scripts/services/Message.js

```diff
@@ -0,0 +1,24 @@
+(function(){
+    function Message($firebaseArray) {
+        
+        var Message = {};
+        
+        // create a reference to Firebase database
+        // then using child() method query messages object
+        var ref = firebase.database().ref().child("messages");
+        
+        // create array of messages
+        var messages = $firebaseArray(ref);
+        
+        //get message object with a specific roomId
+        Message.getByRoomId = function(roomId) {
+            return $firebaseArray(ref.orderByChild('roomId').equalTo(roomId));

```

Comment:
I still believe `orderByChild` belongs on your `var ref` because having it here means every time you get messages for a room firebase has to do another `orderBy` and then search for the `roomId`. Since we will always want messages ordered by `roomId` it is far more efficient to include that in `ref` and then just do `equalTo` here. Your call but something to consider.